python/nav/metrics/data.py

```diff
@@ -110,39 +110,55 @@ def get_metric_data(target, start="-5min", end="now"):
     if isinstance(end, datetime):
         end = end.strftime('%H:%M%Y%m%d')
 
-    query = {
-        'target': target,
-        'from': start,
-        'until': end,
-        'format': 'json',
-    }
-    query = urlencode(query, True)
-
-    _logger.debug("get_metric_data%r", (target, start, end))
-    req = Request(url, data=query.encode('utf-8'))
-    try:
-        response = urlopen(req)
-        json_data = json.load(codecs.getreader('utf-8')(response))
-        _logger.debug("get_metric_data: returning %d results", len(json_data))
-        return json_data
-    except HTTPError as err:
-        _logger.error(
-            "Got a 500 error from graphite-web when fetching %s" "with data %s",
-            err.url,
-            query,
-        )
-        _logger.error("Graphite output: %s", err.fp.read())
-        raise errors.GraphiteUnreachableError("{0} is unreachable".format(base), err)
-    except URLError as err:
-        raise errors.GraphiteUnreachableError("{0} is unreachable".format(base), err)
-    except ValueError:
-        # response could not be decoded
-        return []
-    finally:
+    def divide_chunks(list, n):
+        for i in range(0, len(list), n):
+            yield list[i : i + n]
+
+    queries = [
+        {
+            'target': target_chunk,
+            'from': start,
+            'until': end,
+            'format': 'json',
+        }
+        for target_chunk in divide_chunks(target, 100)
```

Comment:
While moving responsibility for chunking requests into the `get_metric_data()` function may seem reasonable, since this safeguard is now available for all callers, there a are a couple of issues that need to be thought out first:

1. This implementation does not take into consideration the stated interface of this function. `target` may actually be a string, not a list of strings. That usage would be fully broken by this code.
2. Chunking of requests is already performed by some callers of `get_metric_data()` (through `get_metric_average()`). Now there is a very real possibility of "double chunking"). What should we do in this case? See this example:

https://github.com/Uninett/nav/blob/0bca1bb76f2f1c960d8fa9e521e6a47234f083aa/python/nav/netmap/traffic.py#L120-L122